File: fix_imports.py
#!/usr/bin/env python3
"""
Fix @/app/ import paths to relative paths for Figma Make compatibility
"""

import os
import re
from pathlib import Path

def fix_imports_in_file(file_path):
    """Fix all @/app/ imports in a single file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original = content
        
        # Fix imports based on common patterns
        # UI components: @/app/components/ui/* -> ./ui/*
        content = re.sub(r'from ["\']@/app/components/ui/', r'from "./ui/', content)
        content = re.sub(r'import\(["\']@/app/components/ui/', r'import("./ui/', content)
        
        # Components: @/app/components/* -> ./
        content = re.sub(r'from ["\']@/app/components/', r'from "./', content)
        content = re.sub(r'import\(["\']@/app/components/', r'import("./', content)
        
        # Context: @/app/context/* -> ../context/
        content = re.sub(r'from ["\']@/app/context/', r'from "../context/', content)
        content = re.sub(r'import\(["\']@/app/context/', r'import("../context/', content)
        
        # Utils: @/app/utils/* -> ../utils/
        content = re.sub(r'from ["\']@/app/utils/', r'from "../utils/', content)
        content = re.sub(r'import\(["\']@/app/utils/', r'import("../utils/', content)
        
        # Data: @/app/data/* -> ../data/
        content = re.sub(r'from ["\']@/app/data/', r'from "../data/', content)
        content = re.sub(r'import\(["\']@/app/data/', r'import("../data/', content)
        
        # Config: @/app/config/* -> ../config/
        content = re.sub(r'from ["\']@/app/config/', r'from "../config/', content)
        content = re.sub(r'import\(["\']@/app/config/', r'import("../config/', content)
        
        # Pages: @/app/pages/* -> ../pages/
        content = re.sub(r'from ["\']@/app/pages/', r'from "../pages/', content)
        content = re.sub(r'import\(["\']@/app/pages/', r'import("../pages/', content)
        
        if content != original:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"✅ Fixed: {file_path}")
            return True
        return False
        
    except Exception as e:
        print(f"❌ Error fixing {file_path}: {e}")
        return False

def main():
    print("🔧 Fixing import paths for Figma Make...\n")
    
    directories = [
        'src/app/components',
        'src/app/pages',
        'src/app/context',
        'src/app/utils',
    ]
    
    total_fixed = 0
    
    for directory in directories:
        if not os.path.exists(directory):
            continue
            
        print(f"\n📁 Processing {directory}...")
        
        for root, dirs, files in os.walk(directory):
            for file in files:
                if file.endswith(('.tsx', '.ts')):
                    file_path = os.path.join(root, file)
                    if fix_imports_in_file(file_path):
                        total_fixed += 1
    
    print(f"\n✅ Complete! Fixed {total_fixed} files.")

if __name__ == '__main__':
    main()

File: fix-imports.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

// Recursively find all .ts and .tsx files
function findFiles(dir, fileList = []) {
  const files = fs.readdirSync(dir);
  
  files.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory()) {
      if (!file.startsWith('.') && file !== 'node_modules') {
        findFiles(filePath, fileList);
      }
    } else if (file.endsWith('.ts') || file.endsWith('.tsx')) {
      fileList.push(filePath);
    }
  });
  
  return fileList;
}

// Fix imports in a file
function fixImports(filePath) {
  let content = fs.readFileSync(filePath, 'utf8');
  let changed = false;
  
  // Get the directory of the file to calculate relative paths
  const fileDir = path.dirname(filePath);
  const srcAppDir = path.resolve('./src/app');
  
  // Calculate relative path from file to src/app
  const relativeToSrcApp = path.relative(fileDir, srcAppDir);
  const prefix = relativeToSrcApp || '.';
  
  // Replacement patterns
  const replacements = [
    // UI components (always in same directory or subdirectory)
    [/@\/app\/components\/ui\//g, './ui/'],
    
    // Components (relative to current file)
    [/@\/app\/components\//g, relativeToSrcApp ? `${relativeToSrcApp}/components/` : './components/'],
    
    // Other directories
    [/@\/app\/context\//g, relativeToSrcApp ? `${relativeToSrcApp}/context/` : './context/'],
    [/@\/app\/utils\//g, relativeToSrcApp ? `${relativeToSrcApp}/utils/` : './utils/'],
    [/@\/app\/data\//g, relativeToSrcApp ? `${relativeToSrcApp}/data/` : './data/'],
    [/@\/app\/config\//g, relativeToSrcApp ? `${relativeToSrcApp}/config/` : './config/'],
    [/@\/app\/pages\//g, relativeToSrcApp ? `${relativeToSrcApp}/pages/` : './pages/'],
    [/@\/app\/hooks\//g, relativeToSrcApp ? `${relativeToSrcApp}/hooks/` : './hooks/'],
    [/@\/app\/types\//g, relativeToSrcApp ? `${relativeToSrcApp}/types/` : './types/'],
    [/@\/app\/i18n\//g, relativeToSrcApp ? `${relativeToSrcApp}/i18n/` : './i18n/'],
    [/@\/app\/lib\//g, relativeToSrcApp ? `${relativeToSrcApp}/lib/` : './lib/'],
    [/@\/app\/schemas\//g, relativeToSrcApp ? `${relativeToSrcApp}/schemas/` : './schemas/'],
  ];
  
  for (const [pattern, replacement] of replacements) {
    const newContent = content.replace(pattern, replacement);
    if (newContent !== content) {
      content = newContent;
      changed = true;
    }
  }
  
  if (changed) {
    fs.writeFileSync(filePath, content, 'utf8');
    console.log(`✅ Fixed: ${filePath}`);
    return true;
  }
  
  return false;
}

// Main
console.log('🔧 Fixing @/app/ imports...\n');

const files = findFiles('./src/app');
let fixedCount = 0;

files.forEach(file => {
  if (fixImports(file)) {
    fixedCount++;
  }
});

console.log(`\n✅ Fixed ${fixedCount} files!`);
console.log('\n🔍 Verifying...');

// Verify no @/app/ imports remain
let remainingImports = 0;
files.forEach(file => {
  const content = fs.readFileSync(file, 'utf8');
  if (content.includes('@/app/')) {
    console.log(`⚠️  Still has @/app/ imports: ${file}`);
    remainingImports++;
  }
});

if (remainingImports === 0) {
  console.log('✅ All @/app/ imports have been fixed!');
} else {
  console.log(`⚠️  ${remainingImports} files still have @/app/ imports`);
}
